parsel/selector.py

```diff
@@ -342,6 +354,21 @@ def remove_namespaces(self):
         # remove namespace declarations
         etree.cleanup_namespaces(self.root)
 
+    def remove(self):
+        """
+        Remove matched nodes from the parent element.
+        """
+        try:
+            parent = self.root.getparent()
+        except AttributeError:
+            raise InvalidSelectorError(
+                "The specified selector does not have a parent and could not be removed. "
```

Comment:
Hm, I was thinking it fails because it doesn't have a node, not because it doesn't have a parent. But it seems both can be a problem. What happens if you try to call .remove() on a selector for a root node?